api/controllers/StatsController.js

```diff
@@ -121,18 +121,30 @@ function addMatchToRankings(season, match, player, opponent) {
  * }
  */
 function transformSeasonToDTO(season) {
-  const { rankings, ...rest } = season;
+  let { rankings, gameCounts, uniquePlayersPerWeek, ...rest } = season;
+
   // Convert rankings from Map to Array
-  const rankingsAsArray = Array.from(rankings.values()).map((player) => {
+  rankings = Array.from(rankings.values()).map((player) => {
     return {
       ...player,
       matches: Object.fromEntries(player.matches), // Convert matches from Map to Object
     };
   });
+
+  // Remove zero-valued gameCounts + uniquePlayersPerWeek from the end
+  if (gameCounts[gameCounts.length - 1] === 0) {
+    gameCounts.pop();
+    uniquePlayersPerWeek.pop();
+  }
+
+  // Convert uniquePlayersPerWeek from Set[] to int[] number of players
+  uniquePlayersPerWeek = uniquePlayersPerWeek.map((playerSet) => playerSet.size);
```

Comment:
Changing to using `let` instead of `const` and mutating the season in-place has the disadvantage of no longer being a pure function that returns new data w/o side effects, because the season you input here gets mutated in-place, in addition to the data being returned. But it's also more concise, so I think that it's reasonable, especially given that the only intended use of this function is right before returning to the client.

Reply:
I still think the const makes more sense. Changing the shape of what you send in is kinda weird and feels like an anti pattern. Making a new named const and returning that instead is a little less brittle imo. Consider if this fails, you’ll have a half modified season since it works by ref, whereas if you used a new const it wouldn’t have any side effects outside of this function as far as I can tell.